Feature/#94258 translation for dynamic webchat texts (#108)
Cognigy/Webchat: docs/embedding.md

#### Custom Translations

With our latest release (v3.22.0), it is now possible to interpolate dynamic values into your custom translations.

#### Example

```json
{

> 	"textWithButtons": "{text}, Verfügbare Optionen: {buttons}"

Reviewer: This example is now obsolete

| fullSizeImageViewerTitle  | string   | "Full-size image viewer"              | This text is used for the heading in the full-size image viewer.                                                                           |
| downloadFullsizeImage     | string   | "Download full-size image"            | This text is used for the download button in the full-size image viewer.                                                                   |
| closeFullsizeImageModal   | string   | "Close full-size image viewer"        | This text is used for the close button in the full-size image viewer.                                                                      |
| datePickerPreviousMonth   | string   | "Previous month"                      | This text is used as the label for the button to go to the previous month in the date-picker.                                              |
| datePickerNextMonth       | string   | "Next month"                          | This text is used as the label for the button to go to the next month in the date-picker.                                                  |

> | buttonGroupLabel          | string   | "Available actions:"                  | This text is used for live message announcements of text with buttons and quick replies messages.                                          |

Reviewer: Description should be updated as this setting is now also used for button groups inside a gallery item

Author: Modified this as wel

| slidesCountText           | string   | "{slidesCount} slides."               | This text is used for live message announcements of slides count in a gallery message.                                                     |

> | slide                     | string   | "Slide"                               | This text is used for live message announcements of text "Slide" in gallery message.                                                       |

Reviewer: Would be nice if it is made clear where this 'slide' text is used, since it is not obvious with the setting name or value (Used when multiple slides are there in a gallery and to identify the slide position)

Author: I modified the description to indicate the usage

> | listItemGroupLabel        | string   | "Available list items:"               | This text is used for live message announcements of text in list message.                                                                  |

Reviewer: Used for the announcement of the titles of list item in a list message

| Name | Type | Default | Description |
| ---------------------------- | ------ | ------- | ----------- |
| withTranscriptAndCaptions | string | "A video with transcript and captions." | This text is used by video elements with transcript and captions |
| withTranscript | string | "A video with transcript." | This text is used by video messages with transcript |
| withCaptions | string | "A video with captions." | This text is used by video messages with only captions |
| withoutTranscriptAndCaptions | string | "A video message." | This text is used by video messages without transcript and captions |

##### Audio Content

Audio content related text announcements for audio messages
| Name | Type | Default | Description |
| ----------------- | ------ | ------- | ----------- |
| withTranscript | string | "An audio message with transcript" | This text is used by audio messages with transcript |
| withoutTranscript | string | "An audio message." | This text is used by audio messages without transcript |

##### File Content

File content related text announcements for file attachments in messages
| Name | Type | Default | Description |
| ------------- | ------ | ------- | ----------- |

> | singleFile | string | "A {type} named '{fileName}' with size {sizeLabel}." | This text is for single file attachments |

Reviewer: The type will still be in English

Author: Is that fine for now ? or do we want to provide setting to translate file and image ?

Reviewer: Let us leave it as it is. Or we could always announce it as 'file' even if it is an image

| downloadFullsizeImage     | string   | "Download full-size image"                         | This text is used for the download button in the full-size image viewer.                                                 |
| closeFullsizeImageModal   | string   | "Close full-size image viewer"                     | This text is used for the close button in the full-size image viewer.                                                    |
| datePickerPreviousMonth   | string   | "Previous month"                                   | This text is used as the label for the button to go to the previous month in the date-picker.                            |
| datePickerNextMonth       | string   | "Next month"                                       | This text is used as the label for the button to go to the next month in the date-picker.                                |
| textWithButtons           | string   | "{text}. Available options: {buttons}"             | This text is used for live message announcements of text with buttons and quick replies messages.                        |

> | galleryContent            | string   | "Available actions: {buttonlabels}"                | This text is used for live message announcements of text in image gallery.                                               |

Reviewer: Typo in 'buttonLabels'

> | galleryContent            | string   | "Available actions: {buttonlabels}"                | This text is used for live message announcements of text in image gallery.                                               |

Reviewer: I think we could simplify these settings by leaving the keys out and letting the customer to just customize the button-group-identifier texts like 'Available options' (Same could also be used in gallery instead of 'Available actions')

Author: I don't follow you

> | slidesCountText           | string   | "{slidesCount} slides."                            | This text is used for live message announcements of text slides count in image gallery.                                  |

Reviewer: ```suggestion
| slidesCountText           | string   | "{slidesCount} slides."                            | This text is used for live message announcements of slides count in a gallery message.                                  |
```

> | slidesContent             | string   | "Slide {index}: {slideText}. {galleryContent}"     | This text is used for live message announcements of text in slides inside image gallery.                                 |

Author: `galleryContent` is needed as it can be translated

> | listContent               | string   | "{headerLabel} Available list items: {itemLabels}" | This text is used for live message announcements of text in list elements.                                               |

Reviewer: Similar to 'Available options:' text, we could simply leave out headerLabel and itemLabels keys and rename the setting to something like 'listItemGroupLabel'

| imageContent              | `Object` | [See Image Content](#image-content)                | Image related aria labels                                                                                                |
| videoContent              | `Object` | [See Video Content](#video-content)                | Video related aria labels                                                                                                |
| audioContent              | `Object` | [See Audio Content](#audio-content)                | Audio related aria labels                                                                                                |

> | fileContent               | `Object` | [See File content](#file-content)                  | File content related aria labels                                                                                         |

Reviewer: These are not aria-labels. Simply refer them as screen-reader texts

| messageHeader             | `Object` | [See Message Header](#message-header)              | Labels for message headers in conversation                                                                               |
| audioTimeRemaining        | string   | "{time} remaining"                                 | This text is used by aria-valuetext attribute of input element indicating the remaning time left in the Audio playback   |

##### Image Content

Image content related text announcements for image elements
| Name | Type | Default | Description |
| --------------- | ------ | ------- | ----------- |
| downloadable | string | "An image with download option. {altTextLabel}" | This text indicates an image which is downloadable. `altText` is based on the image metadata |

> | nonDownloadable | string | "An image. {altTextLabel}" | This text indicates an image which is not downloadable.`altText` is based on the image metadata |

Reviewer: For these 2, I think we can just ignore the altTextLabel key from the setting

| Name | Type | Default | Description |
| ---------------------------- | ------ | ------- | ----------- |
| withTranscriptAndCaptions | string | "A video with transcript and captions." | This text is used by video elements with transcript and captions |
| withTranscript | string | "A video with transcript." | This text is used by video elements with transcript |
| withCaptions | string | "A video with captions." | This text is used by video elements with only captions |
| withoutTranscriptAndCaptions | string | "A video message." | This text is used by video elements without transcript and captions |

##### Audio Content

Audio content related text announcements for audio elements
| Name | Type | Default | Description |
| ----------------- | ------ | ------- | ----------- |
| withTranscript | string | "An audio message with transcript" | This text is used by audio element with transcript |
| withoutTranscript | string | "An audio message." | This text is used by audio element without transcript |

##### File Content

File content related text announcements for file attachments
| Name | Type | Default | Description |
| ------------- | ------ | ------- | ----------- |

> | singleFile | string | "{text}. A {type} named '{fileName}' with size {sizeLabel}." | This text is for single file attachments |

Reviewer: We could leave out the text key from here. Also, there is a small issue with 'type'. Type can either be 'image' or 'file' (hardcoded in English). So, this cannot be translated

Reviewer: Still not resolved

| multipleFiles | string | "File {index}: '{fileName}', size {sizeLabel}." | This text is used for multiple file attachments |

##### Message Header

Message header related accessibility text content used by screen reader in message header
| Name | Type | Default | Description |
| --------- | ------ | ------- | ----------- |
| user | string | "You said:" | The screen reader text used for announcing user messages |

> | bot | string | "{name} said:" | The screen reader text used for announcing bot message header. The name is the name of the configured configured in the endpoint editor |

Reviewer: The description is weird here

> The following table defines the default aria labels used throughout the Webchat for accessibility. Use them for managing aria-labels based on the preferred language

Reviewer: ```suggestion
The following table defines the default texts used throughout the Webchat for accessibility. Use them for managing aria-labels and other screen reader relevant texts based on the preferred language.

```

| downloadFullsizeImage     | string   | "Download full-size image"                         | This text is used for the download button in the full-size image viewer.                                                 |
| closeFullsizeImageModal   | string   | "Close full-size image viewer"                     | This text is used for the close button in the full-size image viewer.                                                    |
| datePickerPreviousMonth   | string   | "Previous month"                                   | This text is used as the label for the button to go to the previous month in the date-picker.                            |
| datePickerNextMonth       | string   | "Next month"                                       | This text is used as the label for the button to go to the next month in the date-picker.                                |
| actionButton              | string   | "{position} of {total}: {buttonTitleWithTarget}"   | This text is used for indicating list of action buttons in conversation. Position indicates the position of the button   |

> | textWithButtons           | string   | "{text}. Available options: {buttons}"             | This text is used by Aria live attribute for buttons with text                                                           |

Reviewer: ```suggestion
| textWithButtons           | string   | "{text}. Available options: {buttons}"             | This text is used for live message announcements of text with buttons and quick replies messages                                                           |
```

Author: Thanks . For any live region texts such as these this phrasing is applicable right ?
